### PR TITLE
Add solutions and instructions to accessible crossword page

### PR DESCRIPTION
**Pull request by bryophyta**
Add solutions and instructions to accessible crossword page.

Credit to Chris, the Guardian crossword user who brought the lack of solutions to our attention and who provided extremely helpful feedback on the current implementation. Thanks as well to @Chloe-Guardian for help and feedback.

## What is the value of this and can you measure success?

A user brought it to our attention that the text-only version of our crosswords, which is intended to be accessible for screen-reader users, does not contain solutions to the crossword.

Given that we provide solutions for (most) puzzles in the interactive player (e.g. when pressing 'reveal all') this represents a clear lack of parity between the core experiences offered by the text-only and interactive versions.

## What does this change?

* Adds a disclosure element which contains two tables of solutions: one for across clues and one for down clues, to mirror the way we structure the clue.
* Also adds a brief description of what's on the page, based on user feedback that it wasn't self-evident before.

## Screenshots

Three main states:

### 1. Initial render

| Before      | After      |
|-------------|------------|
| Puzzle title and three lists: blanks, across clues, down clues |As before, but the lists are under a heading 'Puzzle description' which includes a description of the puzzle structure. Below it is another heading 'Solutions', which has tables for across and down clues, including their solutions.|

### 2. Solutions disclosure in 'expanded' state

<details>
    <summary>Screenshot <strong>nb. this includes spoilers for Quick 16,000</b></summary>
    image
</details>

### 3. Solutions not available

image

## Checklist

- [x] Tested locally, and on CODE if necessary
- [x] Will not break dotcom-rendering
- [ ] Will not break our database – if updating CAPI, [updated and committed the database files](https://github.com/guardian/frontend/blob/main/docs/03-dev-howtos/15-updating-test-database.md)
- [ ] Meets our accessibility [standards](https://github.com/guardian/recommendations/blob/e647ef695199ea3116ea20d827ef0f1364270a39/accessibility.md)
  - [x] [Tested with screen reader](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#screen-reader)
  - [x] [Navigable with keyboard](https://github.com/guardian/accessibility/blob/main/people-and-technology/02-physical.md#Keyboard)
  - [x] [Colour contrast passed](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#colour)

### CODE REVIEW

File: applications/app/views/fragments/crosswords/accessibleCrosswordContent.scala.html [hunk 1]
```diff
@@ -28,6 +34,61 @@ <h3 class="crossword__clues-header">Down</h3>
...
+                                <button data-component="crossword-solutions-disclosure-trigger" aria-expanded="false" aria-controls="solution-panel" class="button button--medium button--primary button--show-more collection__show-more modern-visible">
+                                    <svg width="12" height="8" aria-hidden="true" viewBox="-2 -2 14 12">
+                                        <g fill="none" transform="rotate(-90 6 4)">
+                                            <path fill="currentColor" d="M1.41.59l4.59 4.58 4.59-4.58 1.41 1.41-6 6-6-6z" />
+                                            <path d="M-6-8h24v24h-24z" />
+                                        </g>
+                                    </svg>
+                                    <span data-component="crossword-solutions-disclosure-trigger-label">Reveal solutions</span>
+                                </button>
+                                <div data-component="crossword-solutions-disclosure-panel" hidden>
```
[the remark this answers is not in the document]
Comment (bryophyta) [reply applications/app/views/fragments/crosswords/accessibleCrosswordContent.scala.html:51]: absolutely! thanks for spotting that 👍

File: applications/app/views/fragments/crosswords/crosswordMetaHeader.scala.html [hunk 2]
```diff
@@ -35,14 +35,6 @@ <h1 itemprop="headline" class="content__headline js-score">@Html(crosswordPage.i
...
             }
 
-            <div class="hide-from-leftcol crossword__clues-header">
-                <a href="https://app.adjust.com/16xt6hai" data-link-name="crossword-mobile-link">Download the Guardian app</a> for a better puzzles experience
-            </div>
-
-            <div class="hide-until-leftcol crossword__clues-header">
-                <a href="https://app.adjust.com/16xt6hai" data-link-name="crossword-desktop-link">Download the Guardian app</a> for a better puzzles experience
-            </div>
-
```
Comment (SiAdcock) [inline applications/app/views/fragments/crosswords/crosswordMetaHeader.scala.html:45]: Why are we removing the ad for the app?

Comment (bryophyta) [reply applications/app/views/fragments/crosswords/crosswordMetaHeader.scala.html:45]: ah, great point, thanks!

I actually don't know how accessible the app versions are (I believe that Android and iOS have different players, too) but it seemed odd to me that we'd point users to the app from this page, unless we are specifically saying that the app is more accessible.

that said, strictly speaking it's outside the scope of this issue so I could just restore it to both pages, or have it toggled by `gridVisable` as you suggest above.

any thoughts on which is preferable? I don't know who 'owns' this link as such so not sure whether it's something I have discretion to remove or not..

Comment (bryophyta) [reply applications/app/views/fragments/crosswords/crosswordMetaHeader.scala.html:45]: That sounds good to me. I've put the app ad back in, but only when `gridVisible`, as you suggested. I also took the opportunity to rename the variable while I was at it.

image
image

**Review by SiAdcock — APPROVED** [review]
Loving it, thanks @bryophyta! 🙏

**Comment by bryophyta** [comment]
> Loving it, thanks @bryophyta! 🙏

thanks for the review!

File: applications/app/views/fragments/crosswords/crosswordMetaHeader.scala.html (hunk 2 above)
Comment (mchv) [reply applications/app/views/fragments/crosswords/crosswordMetaHeader.scala.html:45]: Great work on this @bryophyta. I have added a small follow-up #27541 to hide the download app link on desktop.

**Pull request merged**